src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -332,50 +371,81 @@ private void unpack(File outputFile) throws NoSuchArchiverException {
     }
 
 
-    private void doGet(File outputFile) throws Exception {
-        String urlStr = this.uri.toString();
-        String[] segments = urlStr.split("/");
-        String file = segments[segments.length - 1];
-        String repoUrl = urlStr.substring(0, urlStr.length() - file.length() - 1);
-        Repository repository = new Repository(repoUrl, repoUrl);
-
-        Wagon wagon = this.wagonManager.getWagon(repository.getProtocol());
+    private void doGet(final File outputFile) throws Exception {
+        final RequestConfig requestConfig;
         if (readTimeOut > 0) {
-            wagon.setReadTimeout(readTimeOut);
             getLog().info(
-                "Read Timeout is set to " + readTimeOut + " milliseconds (apprx "
-                    + Math.round(readTimeOut * 1.66667e-5) + " minutes)");
-        }
-        ConsoleDownloadMonitor downloadMonitor = null;
-        if (this.session.getSettings().isInteractiveMode()) {
-            downloadMonitor = new ConsoleDownloadMonitor(this.getLog());
-            wagon.addTransferListener(downloadMonitor);
+                    "Read Timeout is set to " + readTimeOut + " milliseconds (apprx "
+                            + Math.round(readTimeOut * 1.66667e-5) + " minutes)");
+            requestConfig = RequestConfig.custom()
+                    .setConnectTimeout(readTimeOut)
+                    .setSocketTimeout(readTimeOut)
+                    .build();
+        } else {
+            requestConfig = RequestConfig.DEFAULT;
         }
 
-        AuthenticationInfo authenticationInfo = new AuthenticationInfo();
+        final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+
         if (StringUtils.isNotBlank(username)) {
             getLog().debug("providing custom authentication");
             getLog().debug("username: " + username + " and password: ***");
-            authenticationInfo.setUserName(username);
-            authenticationInfo.setPassword(password);
+
+            credentialsProvider.setCredentials(
```

Comment:
@ok2c we could create here UsernamePasswordCredentials only and then create and configure `BasicCredentialsProvider` only ones after `if` block